src/device-registry/bin/www

```diff
@@ -67,37 +104,61 @@ const run = async() => {
             )}`
                 );
             }
-            const measurements = JSON.parse(message.value).data;
-            if (!Array.isArray(measurements) || isEmpty(measurements)) {
-                logger.error(
-                    `KAFKA: the sent measurements are not an array or they are just empty (undefined) --- ${JSON.stringify(
-              measurements
-            )}`
-                );
-            }
-            const responseFromInsertMeasurements = await createEvent.insert(
-                "airqo",
-                measurements
-            );
+            try {
+                // logObject("JSON.parse(message.value.toString()", JSON.parse(message.value))
+                const measurements = JSON.parse(message.value).data;
+                // const measurements = JSON.parse(message.value);
+                if (!Array.isArray(measurements) || isEmpty(measurements)) {
+                    logger.error(
+                        `KAFKA: the sent measurements are not an array or they are just empty (undefined) --- ${JSON.stringify(
+                  measurements
+                )}`
+                    );
+                }
+
+                const { error, value } = eventsSchema.validate(measurements);
+
+                if (error) {
+                    logObject("error.details[0].message", error.details[0].message)
+                    logger.error(
+                        `KAFKA: there is an input validation error --- ${error.details[0].message}`
+                    );
+                } else {
+                    logObject("value", value)
+                    logObject("measurements", measurements)
+                    const request = {
+                        body: measurements
+                    }
+                    const responseFromInsertMeasurements = await createEvent.create(
```

Comment:
Using another util method (`create`) which is capable of receiving the `original` KAFKA messages. @BenjaminSsempala